src/dfx/src/commands/deps/deploy.rs

```diff
@@ -0,0 +1,121 @@
+use crate::lib::deps::{
+    get_canister_prompt, get_pull_canister_or_principal, get_pull_canisters_in_config,
+    get_pulled_wasm_path, load_init_json, load_pulled_json, validate_pulled, InitJson,
+};
+use crate::lib::environment::Environment;
+use crate::lib::error::DfxResult;
+use crate::lib::root_key::fetch_root_key_if_needed;
+use crate::lib::state_tree::canister_info::read_state_tree_canister_controllers;
+
+use anyhow::{anyhow, bail};
+use candid::Principal;
+use clap::Parser;
+use fn_error_context::context;
+use ic_agent::Agent;
+use ic_utils::interfaces::{management_canister::builders::InstallMode, ManagementCanister};
+use slog::{info, Logger};
+
+/// Deploy pulled dependencies.
+#[derive(Parser)]
+pub struct DepsDeployOpts {
+    /// Specify the canister to deploy. You can specify its name (as defined in dfx.json) or Principal.
+    /// If not specified, all pulled canisters will be deployed.
+    canister: Option<String>,
+}
+
+pub async fn exec(env: &dyn Environment, opts: DepsDeployOpts) -> DfxResult {
+    let logger = env.get_logger();
+    let pull_canisters_in_config = get_pull_canisters_in_config(env)?;
+    if pull_canisters_in_config.is_empty() {
+        info!(logger, "There are no pull dependencies defined in dfx.json");
+        return Ok(());
+    }
+
+    let project_root = env.get_config_or_anyhow()?.get_project_root().to_path_buf();
+    let pulled_json = load_pulled_json(&project_root)?;
+    validate_pulled(&pulled_json, &pull_canisters_in_config)?;
+
+    let init_json = load_init_json(&project_root)?;
```

Comment:
If you run `dfx deps pull` and then `dfx deps deploy`, the error looks like below.  This should instead instruct the developer to run `dfx deps init`.

```
Trace mode enabled. Lots of logs coming up.
Error: Failed to read init.json
Caused by: Failed to read init.json
  Failed to read JSON file: Failed to read /Users/ericswanson/w/pulls/deps/init.json: No such file or directory (os error 2)
```